src/server/pkg/ppsdb/ppsdb.go

```diff
@@ -0,0 +1,50 @@
+// Package ppsdb contains the database schema that PPS uses.
+package ppsdb
+
+import (
+	"path"
+
+	etcd "github.com/coreos/etcd/clientv3"
+
+	"github.com/pachyderm/pachyderm/src/client/pps"
+	col "github.com/pachyderm/pachyderm/src/server/pkg/collection"
+)
+
+const (
+	pipelinesPrefix = "/pipelines"
+	jobsPrefix      = "/jobs"
+)
+
+var (
+	// JobsPipelineIndex maps pipeline to jobs started by the pipeline
+	JobsPipelineIndex = col.Index{"Pipeline", false}
+
+	// JobsInputIndex maps job inputs (repos + pipeline version) to output
+	// commit. This is how we know if we need to start a job.
+	JobsInputIndex = col.Index{"Input", false}
+
+	// JobsInputsIndex maps 1.4.5 and earlier style job inputs (repos +
+	// pipeline version) to output commit. This is how we know if we need
+	// to start a job Needed for legacy compatibility.
+	JobsInputsIndex = col.Index{"Inputs", false}
+)
+
+// Pipelines returns a Collection of pipelines
+func Pipelines(etcdClient *etcd.Client, etcdPrefix string) col.Collection {
+	return col.NewCollection(
+		etcdClient,
+		path.Join(etcdPrefix, pipelinesPrefix),
+		[]col.Index{},
+		&pps.PipelineInfo{},
+	)
+}
+
+// Jobs returns a Collection of jobs
+func Jobs(etcdClient *etcd.Client, etcdPrefix string) col.Collection {
+	return col.NewCollection(
+		etcdClient,
+		path.Join(etcdPrefix, jobsPrefix),
+		[]col.Index{JobsPipelineIndex, JobsInputIndex},
+		&pps.JobInfo{},
+	)
+}
```

Comment:
I actually think it would make a lot of sense to relocate `server/pkg/worker` into `server/pps/server` seeing as `server/pps/server` is supposed to be the spot for all servers needed to run pps which is what worker is. It would probably lead to a decent amount of code shared between the 2 packages being made private which would be a nice benefit.